frontend/src/pages/Chapters.tsx

```diff
@@ -30,6 +33,9 @@ const ChaptersPage = () => {
       icon: <FontAwesomeIconWrapper icon="fa-solid fa-right-to-bracket" />,
       url: chapter.idx_url,
     }
+    if (error) {
+      return <ErrorDisplay error={error} onRetry={error.action === 'retry' ? retry : undefined} />
```

Comment:
As the tests are failing, adding a basic test for each of the pages for the error display component should work.

Reply:
yes i writing this test case but one test is failed just resolving that 
![image](https://github.com/user-attachments/assets/51651da7-8806-4813-b69a-2c4a8c2f6639)